sdk/metric/internal/exemplar/rand_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exemplar
+
+import (
+	"context"
+	"math"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFixedSize(t *testing.T) {
+	t.Run("Int64", ReservoirTest[int64](func(n int) (Reservoir[int64], int) {
+		return FixedSize[int64](n), n
+	}))
+
+	t.Run("Float64", ReservoirTest[float64](func(n int) (Reservoir[float64], int) {
+		return FixedSize[float64](n), n
+	}))
+}
+
+func TestFixedSizeSamplingCorrectness(t *testing.T) {
```

Comment:
First of all, I want to share that I am impressed with this unit test 👍 

This test needs description (probability theory / statistics calculations?) on how does this tests checks the correctness.

Reply:
This is creating an exponential distribution of data which is sampled by the reservoir. The mean of the sampled data is evaluated to match the original distribution to show the sampling does not alter the distribution (i.e. no bias towards the start or end).

I can add this description as a comment, what do you think?

I'm not sure adding much more would be helpful. Testing statistical properties is a complex topic and I'm not sure providing a lecture on the subject in the comments is going to help.

Reply:
I find the description very helpful and good enough 👍

> providing a lecture on the subject in the comments is going to help.

This is not needed. Describing what statistical properties are used for testing is good enough. Someone who does not understand the statistics can learn further on his own.